### PR TITLE
feat: We love Liquid Glass

### PR DESCRIPTION
**Pull request by cantpr09ram**
This pull request refactors and improves the widget and view logic for course and student ID display, focusing on better handling of "not logged in" states, code simplification, and improved test coverage. The changes also update project file memberships for better modularity.

**Widget and View Improvements:**

* Unified and enhanced the "not logged in" UI across widgets (`CoursesNextUpSmallView`, `CoursesNextUpViewLockScreenView`, and `StdIDEntryView`) by adding a consistent icon and message, improving user feedback when SSO is missing. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL24-R62) [[2]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL16-R32) [[3]](diffhunk://#diff-b6434d7f4357be6e551eee496eca295f0f8608882fe47bb12d25e510187c9c64L58-R70)
* Improved the empty courses state with clearer messaging and visuals ("下週見") in both small and lock screen widgets. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL24-R62) [[2]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL16-R32)
* Refined course list rendering in `CoursesNextUpSmallView` for better clarity, including weekday name calculation, icon usage, and visual grouping of courses on the same day. [[1]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL52-R73) [[2]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fR124-R178) [[3]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fL130-L136)

**Logic and Codebase Simplification:**

* Refactored `Provider` in `CoursesWidget.swift` to simplify weekday calculations, remove redundant helper methods, and use a service for next-up course logic. [[1]](diffhunk://#diff-3c6bc3911231e6cd7bed6025a5108c3ae48060c37482c14e8a343397e1c97adcL12-L66) [[2]](diffhunk://#diff-3c6bc3911231e6cd7bed6025a5108c3ae48060c37482c14e8a343397e1c97adcL100-L130)
* Updated previews and test data for more comprehensive widget state coverage, including logged out and edge-case scenarios. [[1]](diffhunk://#diff-471a903e2fe3879dcea6d45aaa7d403a6bd4aa4aff49d53bb07d78121551fa0bL93-R113) [[2]](diffhunk://#diff-d5f08cf431d024bee38f56bc4485d28f6e657c06f608413ef5a2ebe728b00a1fR218-R242) [[3]](diffhunk://#diff-b6434d7f4357be6e551eee496eca295f0f8608882fe47bb12d25e510187c9c64R100-R101)

**Testing Enhancements:**

* Added a new test to ensure that on Sunday afternoons with no remaining courses, the widget displays the next Monday's course, improving reliability for weekly transitions.
* Clarified and adjusted the test for filtering out courses ending within 20 minutes, ensuring correct behavior.

**Project Structure:**

* Updated Xcode project file memberships to reflect the new modular structure, moving course-related utilities into a dedicated folder and ensuring correct test target inclusion. [[1]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838L157-R172) [[2]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838L183-R194) [[3]](diffhunk://#diff-354e30eed182121503a3ab91fc76d3f47817cbd8eefc060b0db627893b6e0838R216-R229)

### CODE REVIEW

**Comment by cantpr09ram** [comment]
hey @copilot can you provide right setting for running xcodebuild on GitHub Actions

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -44,7 +43,10 @@ jobs:
...
-          xcodebuild -project dauphin.xcodeproj \
-            -scheme "dauphin" -configuration Debug \
-            -destination 'platform=iOS Simulator,name=iPhone 15,OS=17.5' \
+          xcodebuild \
+            -project dauphin.xcodeproj \
+            -scheme "dauphin" \
+            -configuration Debug \
+            -destination 'generic/platform=iOS Simulator' \
+            -sdk iphonesimulator \
             clean build
```
Comment (winstonsung) [inline .github/workflows/ci.yml:52]: Undo removed newline at the end of the file.

File: README.md [hunk 2]
```diff
@@ -1,16 +1 @@
 # Dauphin
```
Comment (winstonsung) [inline README.md:1]: Is there any reason to blank the README?

File: App/dauphin.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -791,8 +819,9 @@
 		1857DAF02E001FA0008EFC8E /* Debug */ = {
 			isa = XCBuildConfiguration;
 			buildSettings = {
-				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
+				ASSETCATALOG_COMPILER_APPICON_NAME = AppIconDev;
```
Comment (winstonsung) [inline App/dauphin.xcodeproj/project.pbxproj:800]: Intended?

File: App/dauphin.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -816,11 +845,11 @@
 				"INFOPLIST_KEY_UIStatusBarStyle[sdk=iphonesimulator*]" = UIStatusBarStyleDefault;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations = UIInterfaceOrientationPortrait;
 				INFOPLIST_KEY_UISupportedInterfaceOrientations_iPad = "UIInterfaceOrientationLandscapeLeft UIInterfaceOrientationLandscapeRight UIInterfaceOrientationPortrait UIInterfaceOrientationPortraitUpsideDown";
-				IPHONEOS_DEPLOYMENT_TARGET = 16.6;
+				IPHONEOS_DEPLOYMENT_TARGET = 17.6;
```
Comment (winstonsung) [inline App/dauphin.xcodeproj/project.pbxproj:826]: Just to ensure: there are changes that make the app unusable in iOS 16.6, right?

File: App/dauphin/Localizable.xcstrings [hunk 5]
```diff
@@ -286,14 +289,12 @@
...
         }
       }
-    },
-    "Done" : {
-
     },
     "Enjoy your free day!" : {
 
     },
     "First" : {
```
Comment (winstonsung) [inline App/dauphin/Localizable.xcstrings:296]: Remove stale strings.

File: App/dauphin/Localizable.xcstrings [hunk 6]
```diff
@@ -594,6 +599,9 @@
     },
     "Your student ID is %@" : {
 
+    },
+    "校務行事曆" : {
```
Comment (winstonsung) [inline App/dauphin/Localizable.xcstrings:603]: Fix source string: should in English.

File: App/dauphin/Utilities/Letter2Coordinate.swift [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+    self.id = id
+    self.code = code
+    self.name = name
+    self.coordinate = coordinate
+
+  }
+}
+
+let letterLocations: [String: L2GData] = [
+  "X": L2GData(
```
Comment (winstonsung) [inline App/dauphin/Utilities/Letter2Coordinate.swift:19]: Location code "X" is actually located/assigned to "Five-tiger Hill Parking Lot".

Suggestion: location code "ZZZ" is preserved for "other location".

File: App/dauphin/Utilities/Letter2Coordinate.swift [hunk 8]
```diff
@@ -0,0 +1,160 @@
...
+    coordinate: .init(latitude: 25.1756, longitude: 121.449)
+  ),
+  "K": L2GData(
+    code: "K",
+    name: "建築館",
+    coordinate: .init(latitude: 25.17644, longitude: 121.45102)
+  ),
+  "Q": L2GData(
+    code: "Q",
+    name: "教育館",
```
Comment (winstonsung) [inline App/dauphin/Utilities/Letter2Coordinate.swift:61]: No longer "教育館", use "傳播Q館" instead.

File: App/dauphin/Utilities/Letter2Coordinate.swift [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+    coordinate: .init(latitude: 25.1754, longitude: 121.448)
+  ),
+  "J": L2GData(
+    code: "J",
+    name: "麗澤國際學舍",
+    coordinate: .init(latitude: 25.1763, longitude: 121.448)
+  ),
+  "O": L2GData(
+    code: "O",
+    name: "傳播館",
```
Comment (winstonsung) [inline App/dauphin/Utilities/Letter2Coordinate.swift:51]: "傳播O館".

File: App/dauphin/View/CourseSchedule/Day/DayScheduleView.swift [hunk 10]
```diff
@@ -112,7 +51,7 @@ struct DayScheduleView: View {
               .font(.system(size: 60))
               .foregroundColor(.secondary)
 
-            Text("No courses for \(dates[selectedDateIndex].weekday)")
+            Text("No courses for Today")
```
Comment (winstonsung) [inline App/dauphin/View/CourseSchedule/Day/DayScheduleView.swift:54]: Format: "No courses for today" or "No Courses for Today".

File: App/dauphin/View/Other/Library/LibraryView.swift [hunk 11]
```diff
@@ -5,7 +5,6 @@
 //  Created by \u8b19 on 11/25/24.
 //
 
-import Code39
```
Comment (winstonsung) [inline App/dauphin/View/Other/Library/LibraryView.swift:8]: 😂
(no further action required for this comment)

File: App/CoursesWidget/Views/NextUpLockScreenView.swift [hunk 12]
```diff
@@ -13,20 +7,32 @@ struct CoursesNextUpViewLockScreenView: View {
...
   var body: some View {
     if entry.ssoStuNo.isEmpty {
-      Text(entry.ssoStuNo.isEmpty ? "尚未登入" : entry.ssoStuNo)
-        .font(.headline)
-        .padding()
+      HStack(spacing: 10) {
+        Image(systemName: "person.text.rectangle.trianglebadge.exclamationmark.fill")
+          .font(.system(size: 40, weight: .semibold))
+
+        Text("尚未登入")
```
Comment (winstonsung) [inline App/CoursesWidget/Views/NextUpLockScreenView.swift:14]: Localizable?

File: App/dauphin.xcodeproj/project.pbxproj (hunk 4 above)
Comment (winstonsung) [reply App/dauphin.xcodeproj/project.pbxproj:826]: ?

**Comment by cantpr09ram** [comment]
@winstonsung 這個要多分幾次修，我先合併再一個一個處理可以嗎？

**Review by winstonsung — APPROVED** [review]
